### DISCUSSION TITLE
Can @prisma/client be a development dependency?

### DISCUSSION THREAD

**Discussion by nemanjam** [prisma/prisma#11178] [Q&A] [answered]
As you can see Prisma dependencies take almost 200MB in my Docker production image. For the sake of simplifying deployment and seeding process I put both `"@prisma/client": "3.8.0"` and `"prisma": "^3.8.0"` as production dependencies and now I am about to refactor that to reduce image size.

```bash
du -sh ./node_modules/* | sort -nr | grep '\dM.*'

123.8M  ./node_modules/@prisma
95.6M   ./node_modules/@next
59.3M   ./node_modules/prisma
41.2M   ./node_modules/react-icons
32.5M   ./node_modules/next
17.1M   ./node_modules/faker
5.0M    ./node_modules/moment
```

This doc clearly says that `"prisma": "^3.8.0"` should be in dev dependencies. 

https://www.prisma.io/docs/concepts/components/prisma-cli/installation

Ok, but what about `"@prisma/client": "3.8.0"`, can it be moved to dev dependencies too? I read this doc but it's difficult to say if after `npx prisma generate` is executed and client is generated `"@prisma/client": "3.8.0"` dependency is still needed or `const prisma = new PrismaClient();` can run without it just based on generated client?

https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/generating-prisma-client

I saw this example how to run `npx prisma migrate deploy` and `npx prisma db seed` in a separate container with it's own Dockerfile and docker-compose.yml but what I don't like in this example is that prisma folder with schema and seed script are copied inside the Dockerfile, meaning I'd have to rebuild it every time schema is changed, I would like to pass schema as bind mount volume. I'd like to keep only  `"@prisma/client": "3.8.0"` and `"prisma": "^3.8.0"` packages inside Dockerfile and pass database url and schema after container starts, with docker-compose.yml. That should alow me to remove those two ~200MB dependencies from my main app production image.

https://github.dev/notiz-dev/nestjs-prisma-starter

Is this possible? If not, do you know of better example how to perform migration and seed and keep dependencies out of production image?

**Comment by samrathchadha** [comment, marked as answer]
The reason it can't be moved is because `@prisma/client` is what you use to access the generated code(created when you run `npx prisma generate`). This could be a feature request though.

**Reply by nemanjam** [reply]
It seems to me that I wont save any space by removing just `"prisma": "^3.8.0"` from production dependencies. I am not sure it's worth trouble having separate image just for migration.

```bash
yarn add prisma
du -hs node_modules
133M	node_modules
---
yarn add @prisma/client
du -hs node_modules
143M	node_modules
---
yarn add prisma @prisma/client
du -hs node_modules
143M	node_modules
```

**Reply by janpio** [reply]
Prisma CLI is not needed in production.
Prisma Client usually as it is used as the proxy to embed the generated Client (which is stored in `.prisma` which is probably a good reason for the additional space usage after you add `@prisma/client`)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
